src/hypothesis/strategies.py

```diff
@@ -901,7 +888,10 @@ def binary(
     """Generates the appropriate binary type (str in python 2, bytes in python
     3).
 
-    min_size, average_size and max_size have the usual interpretations.
+    min_size and max_size have the usual interpretations.
+
+    The average_size argument is deprecated.  Internal upgrades since
+    Hypothesis 1.x mean we no longer needed this hint to generate useful data.
```

Comment:
Not a request for changes, but it would be nice if we had some sort of include mechanism for this sort of message in our docstrings rather than having to copy and paste it.

Maybe lets not open that can of worms though.

Reply:
Experience (#823 and #1119) suggests that such a mechanism is more trouble than it's worth :sob:

Reply:
Or that we need a better mechanism. 😉